osg_configure/configure_modules/gratia.py

```diff
@@ -358,20 +357,20 @@ def _make_subscription(self, probe, probe_file, probe_host, site, hostname):
             probe = 'gridftp-transfer'
 
         try:
-            buf = open(probe_file).read()
+            buf = open(probe_file, "r", encoding="latin-1").read()
```

Comment:
So, what's the deal with specifying `latin-1` specifically?

Reply:
latin-1 is the encoding for 'I want to treat this as bytes but need to pass it to something that expects a str'. Every byte is mapped to the unicode chr at the same index, so you don't get any UnicodeDecodeErrors and it's easy to do a round trip with it.

Only thing to remember is not to run .upper() or .lower() on that string if you want to convert it back, because not all characters in latin-1 have the opposite-case variants in latin-1.

Reply:
Thanks for the explanation, that is wild.

Reply:
I mean, neat.